deebotozmo/cli.py

```diff
@@ -0,0 +1,458 @@
+    context_settings={"ignore_unknown_options": True},
+)
+@click.argument("area", type=click.STRING, required=True)
+@click.argument("cleanings", type=click.INT, required=False)
+@coro
+async def custom_area(area: str, cleanings: int = 1) -> None:
+    """Click subcommand that runs a clean in a custom area."""
+    await run_with_login(CleanCustomArea(map_position=area, cleanings=cleanings))
+
+
+@cli.command(
+    name="spotarea",
+    help='Cleans provided rooms(s), ex: "0,1" | Use GetRooms to see saved numbers',
+    context_settings={"ignore_unknown_options": True},
+)
+@click.argument("rooms", type=click.STRING, required=True)
+@click.argument("cleanings", type=click.INT)
+@coro
+async def spot_area(rooms: str, cleanings: int = 1) -> None:
+    """Click subcommand that runs a clean in a specific room."""
+    await run_with_login(CleanSpotArea(area=rooms, cleanings=cleanings))
+
+
+@cli.command(name="setfanspeed", help="Set Clean Speed")
+@click.argument("speed", type=click.STRING, required=True)
+@coro
+async def set_fan_speed(speed: str) -> None:
+    """Click subcommand that sets the fan speed."""
+    await run_with_login(SetFanSpeed(speed))
+
+
+@cli.command(name="setwaterlevel", help="Set Water Level")
+@click.argument("level", type=click.STRING, required=True)
+@coro
+async def set_water_level(level: str) -> None:
+    """Click subcommmand that sets the water level."""
+    await run_with_login(SetWaterLevel(level))
+
+
+@cli.command(help="Returns to charger")
+@coro
+async def charge() -> None:
+    """Click subcommand that returns to charger."""
+    await run_with_login(Charge())
+
+
+@cli.command(help="Pause the robot")
+@coro
+async def pause() -> None:
+    """Click subcommand that pauses the clean."""
+    await run_with_login(CleanPause())
+
+
+@cli.command(help="Resume the robot")
+@coro
+async def resume() -> None:
+    """Click subcommand that resumes the clean."""
+    await run_with_login(CleanResume())
+
+
+@cli.command(name="getcleanlogs", help="Get Clean Logs")
+@coro
+async def get_clean_logs() -> None:
+    """Click subcommand that returns clean logs."""
+    vacbot = CliUtil()
+    try:
+        await vacbot.before()
+
+        lock = asyncio.Event()
+
+        async def on_clean_event(event: CleanLogEvent) -> None:
+            print(json.dumps(asdict(event)))
+            lock.set()
+
+        lock.clear()
+        listener = vacbot.bot.events.clean_logs.subscribe(on_clean_event)
+        await lock.wait()
+        listener.unsubscribe()
+    finally:
+        await vacbot.after()
+
+
+@cli.command(help="Get robot statuses [Status,Battery,FanSpeed,WaterLevel]")
+@coro
+async def statuses() -> None:
+    """Click subcommand that returns the robot status."""
+    vacbot = CliUtil()
+    try:
+        await vacbot.before()
+        lock = asyncio.Event()
+
+        print(
+            f"Vacuum State: {str(vacbot.bot.status.state).rsplit('.', maxsplit=1)[-1]}"
+        )
```

Comment:
```suggestion
            async def on_status(event: StatusEvent) -> None:
                print(f"Vacuum available: {event.available}")
                print(f"Vacuum State: {event.state.name}")
                
            status_listener = vacbot.bot.events.status.subscribe(on_status)
            await lock.wait()
            status_listener.unsubscribe()
```

now you can remove mqtt as you get the status too.